plugins/metrics/pkg/apis/remoteread/remoteread.proto

```diff
@@ -12,15 +12,54 @@ import "google/api/annotations.proto";
 package remoteread;
 
 service RemoteReadGateway {
-  rpc AddTarget(TargetAddRequest) returns (google.protobuf.Empty);
-  rpc EditTarget(TargetEditRequest) returns (google.protobuf.Empty);
-  rpc RemoveTarget(TargetRemoveRequest) returns (google.protobuf.Empty);
-  rpc ListTargets(TargetListRequest) returns (TargetList);
+  rpc AddTarget(TargetAddRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/add"
```

Comment:
Suggest renaming some of these to be more "RESTful", e.g. 
add: `POST /targets`
edit: `PUT /targets/{name}`
remove: `DELETE /targets/{name}`
list: `GET /targets`